community/index/src/main/java/org/neo4j/index/gbptree/GenSafePointer.java

```diff
@@ -43,11 +43,11 @@
  */
 class GenSafePointer
 {
-    public static final long MIN_GENERATION = 1L;
+    static final long MIN_GENERATION = 1L;
```

Comment:
this class have a lot of static methods that are just helpers and i find it hard to see what it has to do with GenSafePointer (GenerationalPointer), also comment in java doc mentions one return time as a limitation of some sort can we talk about that maybe?